services/f-droid/f-droid.service.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
+
+const BaseHTTPService = require('../base-http')
+const { addv: versionText } = require('../../lib/text-formatters')
+const { version: versionColor } = require('../../lib/color-formatters')
+const { InvalidResponse } = require('../errors')
+
+module.exports = class FDroid extends BaseHTTPService {
+  async fetch({ appId }) {
+    const url = `https://f-droid.org/en/packages/${appId}/`
+    return this._requestHTTP({
+      url,
+      options: {},
+      errorMessages: {
+        404: 'app not found',
+      },
+    }).then(({ res, buffer }) => {
+      const website = buffer.toString()
+      // we assume the layout as provided here:
+      // https://gitlab.com/fdroid/fdroid-website/blob/9ae61894a18889ed749d36d5afbd0db3d0b0cfdd/_layouts/package.html#L147
+      const match = website.match(
+        /<div\s[^>]*class="package-version-header"(?:\s[^>]*)?>[^<]*<a\s+name="([^:>]*)"(?:\s[^>]*)?>/
+      )
+      if (!match) {
+        throw new InvalidResponse({
+          prettyMessage: 'fix this badge',
+          underlyingError: new Error('could not find version on website'),
+        })
+      }
+      return { version: match[1] }
+    })
+  }
+
+  static render({ version }) {
+    return {
+      message: versionText(version),
+      color: versionColor(version),
+    }
```

Comment:
There's a new helper function [`renderVersionBadge`](https://github.com/badges/shields/blob/302c8606ff4c975b6310fe2cdfe58f92cede290b/lib/version.js#L144-L150) which you can use here. You can pass `{ version }` to it; the two other parameters are optional.

Reply:
Could you please update this so it uses the helper function?

Reply:
~Could you please update this so it uses the helper function?~

We're not using the helper function consistently, so this comment can be deferred to its own issue.